WordPress/Classes/ViewRelated/Reader/Manage/ReaderTagsTableViewController.swift

```diff
@@ -0,0 +1,174 @@
+class ReaderTagsTableViewController: UIViewController {
+
+    private let style: UITableView.Style
+
+    private lazy var tableView: UITableView = {
+        let tableView = UITableView(frame: .zero, style: style)
+        tableView.translatesAutoresizingMaskIntoConstraints = false
+        return tableView
+    }()
+
+    private var tableViewHandler: OffsetTableViewHandler?
+
+    init(style: UITableView.Style) {
+        self.style = style
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        view.addSubview(tableView)
+        view.pinSubviewToAllEdges(tableView)
+
+        setupTableHandler()
+    }
+
+    private func setupTableHandler() {
+        let handler = OffsetTableViewHandler(tableView: tableView)
+        handler.delegate = self
+        tableViewHandler = handler
+    }
+}
+
+// MARK: - WPTableViewHandlerDelegate
+
+extension ReaderTagsTableViewController: WPTableViewHandlerDelegate {
+
+    func managedObjectContext() -> NSManagedObjectContext {
```

Comment:
there's a few Core Data-related tasks, which preferably should not be done in a UIViewController. They mainly seem to be related to the WPTableViewHandlerDelegate, so maybe we could extrapolate this delegate in a new object, away from the UIViewController? Not necessarily in this PR

Reply:
> so maybe we could extrapolate this delegate in a new object

I had a few thoughts on this we can discuss tomorrow. I think passing in the managed object context, topic service, etc. make testing easier. Alternatively, I could add some kind of model that implements the `WPTableViewHandlerDelegate` and separates much of this functionality.

Reply:
Sounds good, let's talk tomorrow! This is just a suggestion anyway, and we could certainly address it in a separate PR.